service/history/hsm/tree.go

```diff
@@ -185,22 +212,52 @@ func (n *Node) Path() []Key {
 
 // Outputs returns all outputs produced by transitions on this tree.
 func (n *Node) Outputs() []PathAndOutputs {
-	var paos []PathAndOutputs
-	if len(n.cache.outputs) > 0 {
-		paos = append(paos, PathAndOutputs{Path: n.Path(), Outputs: n.cache.outputs})
+	root := n.root()
+	if root.opLog == nil {
+		return nil
 	}
-	for _, child := range n.cache.children {
-		paos = append(paos, child.Outputs()...)
+
+	currentPath := n.Path()
+
+	if root.opLog.IsDeleted(currentPath) {
+		return nil
 	}
+
+	// Group transitions by their path
+	pathToTransitions := make(map[string][]TransitionOutputWithCount)
+	for _, t := range root.opLog.Transitions {
+		// Only include transitions under current node's path
```

Comment:
And that aren't for deleted nodes. I don't see that covered here.

Reply:
Done